kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java

```diff
@@ -264,6 +254,36 @@ public void abortTransaction() throws ProducerFencedException {
             this.delegate.abortTransaction();
         }
 
+        @Override
+        public void close() {
+            this.delegate.close();
+        }
+
+        @Override
+        public void close(long timeout, TimeUnit unit) {
+            this.delegate.close(timeout, unit);
+        }
+
+        @Override
+        public String toString() {
+            return this.getClass().getSimpleName() + " [delegate=" + this.delegate + "]";
+        }
+    }
+
+    private static final class CloseLazyProducer<K, V> extends ProducerDecorator<K, V> {
```

Comment:
While you're on this part, would you mind renaming this to `LazilyClosedProducer`?

If you're up for the task, a little bit of JavaDoc would be nice too; I am fine with adding this myself later on though.

Reply:
While I am working on this, it occurs to me that what we are using here is not really a cache but an instance pool. What do you think about calling the `BlockingQueue` a `pool` and renaming this class to `PoolableProducer` or maybe `PooledProducer` because its main property is not only that it is closed lazily but also that it returns itself to the pool when `close` is called.

The `CloseNeverProducer` (I never liked that name anyway but couldn't think of something better at the time) could maybe be renamed to `ShareableProducer`, `SingletonProducer` or something like that.

Reply:
Amazing, really like those suggestions @lbilger! I'll resolve this comment and take it with me to #16.